conans/model/info.py

```diff
@@ -617,10 +617,10 @@ def default_std_non_matching(self):
             self.settings.compiler.cppstd = self.full_settings.compiler.cppstd
 
     def shared_library_package_id(self):
-        if self.full_options.shared:
+        if "shared" in self.full_options and self.full_options.shared:
```

Comment:
I needed to change this, as now missing options also raise in ``self.info.options`` fields.